embassy-stm32/src/pka/mod.rs

```diff
@@ -16,67 +16,297 @@ static PKA_WAKER: AtomicWaker = AtomicWaker::new();
 // List of RAM addresses for data. Since embassy implementation of RAM access already moves base address to
 // 0x0400 and multiplies offset by 4 for 32-bit access, we have to take address from reference manual, subtract 0x0400
 // and divide it by 4 to get the correct offset.
+enum MontParamComputation {
+    ModulusLength,
+    Modulus,
+    Result,
+}
+
+impl MontParamComputation {
+    const MODULUS_LENGTH_ADDR: usize = (0x0404 - 0x0400) / 4;
+    const MODULUS_ADDR: usize = (0x0D5C - 0x0400) / 4;
+    const RESULT_ADDR: usize = (0x0594 - 0x0400) / 4;
+
+    fn value(&self) -> usize {
+        match self {
+            MontParamComputation::ModulusLength => Self::MODULUS_LENGTH_ADDR,
+            MontParamComputation::Modulus => Self::MODULUS_ADDR,
+            MontParamComputation::Result => Self::RESULT_ADDR,
+        }
+    }
+}
+
```

Comment:
You can assign value to enum directly https://doc.rust-lang.org/reference/items/enumerations.html#implicit-discriminants

Reply:
It can be even an expression like `(0x0404 - 0x0400) /4`

Reply:
I know, check the PKA mode enum. But if I implement this method, I can just call it whenever I need a usize value, instead of casting it each time.

Reply:
I'm not insisting :p You can consider implementing the `From<MontParamComputation> for usize` trait. Then you can just call `into()` instead of `value()` and it seems more Rusty way for me.